http-client-core/src/main/java/io/micronaut/http/client/AbstractHttpClientFactory.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.client;
+
+import io.micronaut.core.annotation.Creator;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.convert.ConversionService;
+import io.micronaut.http.codec.MediaTypeCodecRegistry;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+/**
+ * Abstract class implementation of {@link HttpClientFactory}.
+ * @author Sergio del Amo
+ * @since 4.0.0
+ */
+@Internal
+public abstract class AbstractHttpClientFactory<T extends HttpClient> implements HttpClientFactory {
+
+    protected final MediaTypeCodecRegistry mediaTypeCodecRegistry;
+    protected final ConversionService conversionService;
+
+    // Used by HttpClient.create(...)
+    protected AbstractHttpClientFactory() {
+        this(null, ConversionService.SHARED);
+    }
+
+    @Creator
```

Comment:
is `@Creator` needed here?

Reply:
Nope, nor was the empty constructor, so I removed both